### PR TITLE
Make sure the agent view is working with the Scheduler

### PR DESCRIPTION
**Pull request by hlloreda**
# Description

`Inmantaci` closed the previous pull request as the target branch was `issue/halt_env_and_pause_agents`
Supersed: https://github.com/inmanta/inmanta-core/pull/8221

Updated a test to make sure the agent view is working with the Scheduler

closes https://github.com/inmanta/inmanta-core/issues/8186

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: src/inmanta/data/dataview.py [hunk 1]
```diff
@@ -1211,16 +1213,24 @@ def get_base_url(self) -> str:
...
+                                    a.environment,
+                                    a.last_failover,
+                                    a.paused,
+                                    a.unpause_on_resume,
+                                    sa_join.hostname as process_name,
+                                    sa_join.process as process_id,
+                                    (CASE WHEN a.paused THEN 'paused'
+                                        WHEN NOT a.paused AND sa_join.id_primary IS NULL THEN 'down'
+                                        ELSE 'up'
+                                    END) as status""",
```
Comment (Hugo-Inmanta) [inline src/inmanta/data/dataview.py:1226]:
```suggestion
                                    (CASE
                                        WHEN a.paused THEN 'paused'
                                        WHEN NOT a.paused AND sa_join.id_primary IS NULL THEN 'down'
                                        ELSE 'up'
                                    END) as status""",
```

File: tests/server/test_agents_api.py [hunk 2]
```diff
@@ -123,7 +123,7 @@ def agent_names(agents: list[dict[str, str]]) -> list[str]:
 async def test_agents_paging(server, client, env_with_agents: None, environment: str, order_by_column: str, order: str) -> None:
     result = await client.get_agents(
         environment,
-        filter={"status": ["paused", "up"]},
```
Comment (Hugo-Inmanta) [inline tests/server/test_agents_api.py:126]: why change this ?

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]

File: src/inmanta/data/dataview.py [hunk 3]
```diff
@@ -1211,16 +1213,24 @@ def get_base_url(self) -> str:
...
+            from_clause=f""" FROM {Agent.table_name()} a
+                             CROSS JOIN (SELECT sa.name, sa.id_primary, ap.hostname, ai.process FROM agent sa
+                             LEFT JOIN {AgentInstance.table_name()} ai ON sa.id_primary=ai.id
+                             LEFT JOIN {AgentProcess.table_name()} ap ON ai.process = ap.sid
+                             WHERE sa.environment = $1 AND sa.name = $2) sa_join """,
```
Comment (Hugo-Inmanta) [inline src/inmanta/data/dataview.py:1231]:
```suggestion
            from_clause=f""" 
            FROM {Agent.table_name()} a
            CROSS JOIN (
                SELECT 
                    sa.name,
                    sa.id_primary,
                    ap.hostname,
                    ai.process
                FROM {Agent.table_name()} sa
                LEFT JOIN {AgentInstance.table_name()} ai ON sa.id_primary=ai.id
                LEFT JOIN {AgentProcess.table_name()} ap ON ai.process = ap.sid
                WHERE sa.environment = $1 AND sa.name = $2
            ) sa_join 

            """,
```

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]

File: tests/server/test_agents_api.py (hunk 2 above)
Comment (hlloreda) [reply tests/server/test_agents_api.py:126]: I refactored a bit to make it obvious in the `env_with_agents` fixture why everything (except paused agents) is down

**Review by Hugo-Inmanta — APPROVED** [review]

**Pull request closed without merging**